module-2/backend/week_two.md

```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+---It's a wrapper for SQL that allows you to easily access a database within an application with easy commands.
+
 2. What is a migration?
+--It's a set of instructions for creating a database.
+
 3. How does a table relate to a model?
+--We have a model for each table.  A model is a class that does calculations and operations on the data.
+
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+--They are class methods, which is why they're in the models which are classes.  In Blogger, we had an Article class and an Attachment class.  Each attachment "belonged_to" an article and an article could "have_many" attachments.
+
 5. What do they allow you to do?
+--They establish relationships between tables and classes.
+
 6. What's the difference between agile workflow and waterfall method?
+--Agile workflow starts with small chunks of functionality and tests and reevaluates each step along the way.  Waterfall does all the planning at first with end goals in mind, then builds it all out, and then tests it all at the end.
+
 7. What is the difference between `#new` and `#create`?
+--Both create a new object but only create saves it in the database.
+
 8. At a basic level, what does cURL allow you to do?
+--It enables you to communicate with servers over the internet from the command line or from within a program.
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+--Many-to-Many.  There will be a students table and a teachers table, each of which are connected through a joined students_teachers table.
+
 10. Define foreign key, primary key, and schema.
+--The primary key is the main identifier in a table.  A foreign key is usually an id from a different table which relates the two.  A schema is basically a map of a database and the relationships between its tables.
+
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+--The foreign key in one table refers to the primary key of another table and establishes a relationship between them.
+
 12. What are the parts of an HTTP response?
+--There is a code (3 digits), head, and body.
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
+--Assert and assert_equal
+
 14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+--If you find yourself repeating the same code over and over, such as creating Payloads in RushHour, either use something like FactoryGirl or simply create your own method to create payloads and simply call the method each test.
```

Comment:
Good ideas for drying up code in general, but in terms of **views** specifically, I would take a look at partials. We'll get into this more tomorrow.